services/inventory/services.go

```diff
@@ -198,11 +201,40 @@ func (ss *ServicesService) AddExternalService(ctx context.Context, params *model
 // Remove removes Service without any Agents.
 //nolint:unparam
 func (ss *ServicesService) Remove(ctx context.Context, id string, force bool) error {
-	return ss.db.InTransaction(func(tx *reform.TX) error {
+	pmmAgentIDs := make(map[string]bool)
+	var reloadPrometheusConfig bool
+
+	if e := ss.db.InTransaction(func(tx *reform.TX) error {
 		mode := models.RemoveRestrict
 		if force {
 			mode = models.RemoveCascade
 		}
+
+		agents, err := models.FindAgents(tx.Querier, models.AgentFilters{ServiceID: id})
+		if err != nil {
+			return err
+		}
+
+		for _, agent := range agents {
+			if agent.PMMAgentID != nil {
+				pmmAgentIDs[pointer.GetString(agent.PMMAgentID)] = true
+			} else {
+				reloadPrometheusConfig = true
+			}
+		}
+
 		return models.RemoveService(tx.Querier, id, mode)
-	})
+	}); e != nil {
+		return e
+	}
+
+	for agentID := range pmmAgentIDs {
```

Comment:
we should `SendSetStateRequest` and `RequestConfigurationUpdate` only if force flag is passed. otherwise agents related to service won't be deleted, which means that there is no changes.

Reply:
fixed